validation/start-block_test.go

```diff
@@ -45,15 +46,39 @@ func testStartBlock(t *testing.T, chain *ChainConfig) {
 	}
 }
 
-func getStartBlockWithRetries(ctx context.Context, systemConfigAddr common.Address, client *ethclient.Client) (uint64, error) {
+func getStartBlockWithRetries(ctx context.Context, systemConfigAddr common.Address, genesisBlock uint64, client *ethclient.Client) (uint64, error) {
 	callOpts := &bind.CallOpts{Context: ctx}
 	systemConfig, err := bindings.NewSystemConfig(systemConfigAddr, client)
 	if err != nil {
 		return 0, err
 	}
 	val, err := Retry(systemConfig.StartBlock)(callOpts)
 	if err != nil {
-		return 0, err
+		// Checking when the contract has been deployed, backward compatibility with SystemConfig without StartBlock()
+		// Fetch the block when the proxy has been deployed via the AdminChanged event from 0x0 to the actual admin
+		logs, err := client.FilterLogs(
+			ctx,
+			ethereum.FilterQuery{
+				FromBlock: big.NewInt(int64(genesisBlock)), // we assume SystemConfig has been deployed closely to genesis (100 blocks around)
+				ToBlock:   big.NewInt(int64(genesisBlock + 100)),
+				Addresses: []common.Address{systemConfigAddr},
+				Topics:    [][]common.Hash{{common.HexToHash("0x7e644d79422f17c01e4894b5f4f588d331ebfa28653d42ae832dc59e38c9798f")}}, //AdminChanged hash
```

Comment:
@emilianobonassi can you link to the old version of the SystemConfig contract? I just want to verify this event and the topic hash used above

Reply:
sure @bitwiseguy 

https://etherscan.io/address/0x58b51fb9FeeD00DD846f91D265Eba3cdd855A413

proxy creation https://etherscan.io/tx/0x77d4a15a096b450c43b3911168c32d453708f2c37abad51d42bdd80e54d308af

using proxy admin init to detect deployment block

Reply:
Confirmed topic hash:
```
> cast keccak "AdminChanged(address,address)"
0x7e644d79422f17c01e4894b5f4f588d331ebfa28653d42ae832dc59e38c9798f
```